behave/features/steps/common_steps.py

```diff
@@ -294,13 +295,46 @@ def make_dir(dir):
         os.makedirs(dir)
 
 
-def check_accessibility(context):
+def check_accessibility(context, step_name):
     # Sleep prevents Axe exceptions.
     # If no logs for Axe, Axe is called too fast when trying to inject javascript.
     time.sleep(0.6)
     axe = Axe(context.helperfunc.driver())
     axe.inject()
     results = axe.run()
     if len(results["violations"]) > 0:
-        axe.write_results(results, f"{context.a11y_reports_dir}/a11y.json")
+        result_format = [
+            dict(
+                step=step_name,
+                url=context.helperfunc.get_url(),
+                violations=results["violations"],
+            )
+        ]
+
+        try:
+            f = open(f"{context.a11y_reports_dir}/a11y.json", "r")
+            result_format = json.load(f) + result_format
+            axe.write_results(result_format, f"{context.a11y_reports_dir}/a11y.json")
+            f.close()
+        except FileNotFoundError:
+            axe.write_results(result_format, f"{context.a11y_reports_dir}/a11y.json")
+
     return len(results["violations"]) == 0
+
+
+def filter_accessibility_report(context):
+    with open(f"{context.a11y_reports_dir}/a11y.json", "r") as f:
+        data = json.load(f)
+
+    results_to_copy = []
```

Comment:
Could we rename `results_to_copy` with something more descriptive? _E.g.: `filtered_issues` (or else)_